### ISSUE TITLE
HTTPS Tests are not working

### ISSUE THREAD

**Issue by saendigPhilip** [libcpr/cpr#771]
## Description
There are a lot of errors in the HTTPS Test Server. While @simon-berger made me aware of some of them, I encountered a few more during my work on updating the test server (see #757).

### Expected Behavior
The certificates and keys are moved to the test binary path, so the relative paths to them are correct. Certificates and keys are loaded properly.

### Actual Behavior
The [paths to the certificates and keys](https://github.com/libcpr/cpr/blob/master/test/ssl_tests.cpp#L115-L116) are wrong. For some reason, the tests succeed anyway.

### Possible Fix
Install the certificates and keys relative to the test build directory.

## Expired client certificate
The client certificate is expired and needs to be re-generated. 

### Possible Fix
Re-generate the client certificate. Also, the period of validity should be generously extended, as its private key is publicly available anyway.

### Expected Behavior
If I understood that correctly, there should be a PKI, in which the root certificate signs the server and the client certificate.

### Actual Behavior
The verification of the certificates failed. Openssl states that they are self-signed.
Verification check of the server certificate:
```console
$ openssl verify -CAfile ca.cer server.cer
C = XX, L = Default City, O = Default Company Ltd
error 18 at 0 depth lookup: self-signed certificate
error server.cer: verification failed
```
Verification check of the client certificate:
```console
$ openssl verify -CAfile ca.cer client.cer
C = XX, L = Default City, O = Default Company Ltd
error 18 at 0 depth lookup: self-signed certificate
C = XX, L = Default City, O = Default Company Ltd
error 10 at 0 depth lookup: certificate has expired
error client.cer: verification failed
```

### Possible Fix
Set up a CA as described in the [corresponding chapter in the OpenSSL cookbook](https://www.feistyduck.com/library/openssl-cookbook/online/openssl-command-line/private-ca.html).
This would not only solve the issue, but it would also make the generation of certificates for future tests easier.

## Context
As mentioned before, I am currently working on upgrading the mongoose HTTP test server to support mongoose 7. I need to fix those bugs anyway as they cause problems in mongoose 7. Therefore, I would like to work on that as well.

If anyone notices any further problems or has other suggestions for fixes, please mention them here, then I will have a look at them :)

**Comment by COM8** [comment]
Thanks for reporting! Yes, this is an issue and should be fixed. Would you like to have a look at it, once you are done with your mongoose update?

**Comment by saendigPhilip** [comment]
Yes, I'll try to solve it.
With mongoose 7, the HTTPS tests fail, probably because of the invalid certificates. So, as I mentioned, this issue needs to be solved in order to solve #757 anyway. That's also why I already started working a bit on it.
